cupyx/scipy/linalg/decomp_lu.py

```diff
@@ -114,6 +162,122 @@ def lu_factor(a, overwrite_a=False, check_finite=True):
     return (a, ipiv)
 
 
+def cupy_split_lu(LU, order='C'):
+    assert LU._c_contiguous or LU._f_contiguous
```

Comment:
It would be nice to reuse LU as L or U in terms of memory usage, but then L and U become F-contiguous matrices, right?  `cupy_laswp` will be applied to matrix L if permute_l is True and `cupy_laswp` is slower when L is F-contiguous (as `cupy_laswp` is implemented in in-place way). That is why L and U are C-contiguous matrices as default.

So I will remove the support for the case when LU is C-contigous, but I'd like to keep the rest of it.